### PR TITLE
Add distn

### PR DESCRIPTION
**Pull request by dajmcdon**
Partially addresses #63 .

1. Requires adding some post-processing layers to add the quantiles.
2. Suggest we remove the `...` args in `layer_predict()` (see #46 ) and prefer explicit quantile addition. This would use the `slather` method to process the output as needed (perhaps depending on the fitted model).

### CODE REVIEW

File: R/layer_residual_quantiles.R [hunk 1]
```diff
@@ -0,0 +1,88 @@
+#' Creates predictions based on residual quantiles
+#'
+#' @param frosting a `frosting` postprocessor
+#' @param ... Unused, include for consistency with other layers.
+#' @param probs numeric vector of probabilities with values in (0,1)
+#'   referring to the desired quantile.
+#' @param symmetrize logical. If `TRUE` then interval will be symmetric.
+#' @param name character. The name for the output column.
+#' @param .flag a logical to determine if the layer is added. Passed on to
```
Comment (ChloeYou) [inline R/layer_residual_quantiles.R:9]: Is `.flag` still preserved or did we agree to remove it?

Comment (dajmcdon) [reply R/layer_residual_quantiles.R:9]: It's there for now, just for consistency with the other `layer_*()` functions. But the plan is to change the name later. (Separate PR).

File: R/dist_quantiles.R [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+  tau <- as_list_of(tau, .ptype = double())
+  args <- vec_recycle_common(x = x, tau = tau)
+  qntls <- as_list_of(map2(args$x, args$tau, new_quantiles))
+  new_vctr(qntls, class = "distribution")
+}
+
+
+# format.dist_quantiles <- function(x, ...) {
+#   sprintf("quantiles[%s]", length(x))
+# }
```
Comment (mgyliu) [inline R/dist_quantiles.R:65]: Was this commented out section left here on purpose? If so, maybe good to document what it's for

Comment (dajmcdon) [reply R/dist_quantiles.R:65]: It was an earlier version that doesn't work correctly. Probably should remove.

**Pull request merged**